src/components/MetadataPanel/index.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect } from "react";
+import { Divider } from "antd";
+import classNames from "classnames";
+
+import { AgentMetadata, MetadataDisplay } from "../../constants/interfaces";
+import { formatFloatForDisplay } from "../../util";
+import { FilledCaret } from "../Icons";
+import { AgentDisplayNode } from "../AgentTree";
+
+import styles from "./style.css";
+
+interface MetadataPanelProps {
+    selectedAgentMetadata: AgentMetadata;
+    uiDisplayData: AgentDisplayNode[];
+}
+
+const MetadataPanel: React.FC<MetadataPanelProps> = ({
+    selectedAgentMetadata,
+    uiDisplayData,
+}) => {
+    const [panelExpanded, setPanelExpanded] = React.useState(false);
+
+    const selectedAgentNotRendered =
+        selectedAgentMetadata.agentType >= uiDisplayData.length;
+
+    useEffect(() => {
+        if (!selectedAgentMetadata.uniqueId) {
+            setPanelExpanded(false);
+        } else {
+            setPanelExpanded(true);
+        }
+    }, [selectedAgentMetadata.uniqueId]);
+
+    const displayValues: Record<keyof AgentMetadata, MetadataDisplay> = {
+        uniqueId: {
+            label: <span>Unique ID</span>,
+            value: () => formatFloatForDisplay(selectedAgentMetadata.uniqueId),
+        },
+        agentType: {
+            label: <span>Agent Type</span>,
+            value: () => uiDisplayData[selectedAgentMetadata.agentType].title,
+        },
+        position: {
+            label: <span>Position</span>,
+            value: () => {
+                const pos = selectedAgentMetadata.position;
+                return (
+                    <>
+                        <div>x = {formatFloatForDisplay(pos.x)}</div>
+                        <div>y = {formatFloatForDisplay(pos.y)}</div>
+                        <div>z = {formatFloatForDisplay(pos.z)}</div>
+                    </>
+                );
+            },
+        },
+        rotation: {
+            label: (
+                <>
+                    Rotation
+                    <br />
+                    (degrees)
+                </>
+            ),
+            value: () => {
+                const rot = selectedAgentMetadata.rotation;
+                return (
+                    <>
+                        <div>x = {formatFloatForDisplay(rot.x)}</div>
+                        <div>y = {formatFloatForDisplay(rot.y)}</div>
+                        <div>z = {formatFloatForDisplay(rot.z)}</div>
+                    </>
+                );
+            },
+        },
+        radius: {
+            label: <span>Radius</span>,
+            value: () => formatFloatForDisplay(selectedAgentMetadata.radius),
+        },
+    };
+
+    const getMetadataRows = () => {
+        const keys = Object.keys(
+            selectedAgentMetadata
+        ) as (keyof AgentMetadata)[];
```

Comment:
I like how you are using `keyof AgentMetadata` both here and in displayValues to make sure this is safe.  The `displayValues[key]` lookups look risky to me but the typescript typings are doing their job.